dGame/dComponents/ActivityComponent.cpp

```diff
@@ -74,6 +68,24 @@ ActivityComponent::ActivityComponent(Entity* parent, int32_t activityID) : Compo
 		}
 	}
 }
+void ActivityComponent::LoadActivityData(const int32_t activityId) {
+	CDActivitiesTable* activitiesTable = CDClientManager::Instance().GetTable<CDActivitiesTable>();
+	std::vector<CDActivities> activities = activitiesTable->Query([activityId](CDActivities entry) {return (entry.ActivityID == activityId); });
+
+	for (CDActivities activity : activities) {
+		m_ActivityInfo = activity;
+		if (static_cast<Leaderboard::Type>(activity.leaderboardType) == Leaderboard::Type::Racing && Game::config->GetValue("solo_racing") == "1") {
```

Comment:
Actually, you may want to pull this GetValue out of the loop

Reply:
Any particular reason?

Reply:
The env reading code that might potentially allocate and it's very unlikely to change between iterations

Reply:
how are you supposed to know whether to use solo racing then?

Reply:
```
bool soloRacing = Game::config->GetValue("solo_racing") == "1";
for (...) {
  if (... && soloRacing) ...
  ...
}
```
is what I mean.

Reply:
Ah that was not clear.  Will change this then.